scripts/rcx_central_overnight_patch_cycle.py

```
@@ -0,0 +1,298 @@
+
+def _git_branch() -> str:
+    try:
+        out = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"], cwd=REPO, text=True)
+        return out.strip()
+    except Exception:
+        return "unknown"
+
+
+def _run_pytest(targets: list[str]) -> tuple[bool, str]:
+    cmd = [sys.executable, "-m", "pytest", *targets, "-q", "--tb=line"]
+    proc = subprocess.run(cmd, cwd=REPO, capture_output=True, text=True, timeout=900, check=False)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)